projects/storefrontlib/src/lib/checkout/guards/delivery-mode-set.guard.spec.ts

```diff
@@ -0,0 +1,75 @@
+import { TestBed } from '@angular/core/testing';
+import { RouterTestingModule } from '@angular/router/testing';
+
+import { DeliveryModeSetGuard } from './delivery-mode-set.guard';
+import { CheckoutConfig } from '../config/checkout-config';
+import { DeliveryMode, CheckoutService } from '@spartacus/core';
+import { of, Observable } from 'rxjs';
+import { UrlTree } from '@angular/router';
+
+const mockCheckoutConfig: CheckoutConfig = {
+  checkout: {
+    steps: [
+      {
+        id: 'deliveryMode',
+        name: 'checkoutProgress.label.deliveryMode',
+        url: '/checkout/delivery-mode',
+        type: ['deliveryMode'],
+      },
+    ],
+  },
+};
+
+const mockDeliveryMode: DeliveryMode = {
+  name: 'test mode name',
+};
+
+class MockCheckoutService {
+  getSelectedDeliveryMode(): Observable<DeliveryMode> {
+    return of();
+  }
+}
+
+describe(`DeliveryModeSetGuard`, () => {
+  let guard: DeliveryModeSetGuard;
+  let mockCheckoutService: MockCheckoutService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        DeliveryModeSetGuard,
+        { provide: CheckoutConfig, useValue: mockCheckoutConfig },
+        { provide: CheckoutService, useClass: MockCheckoutService },
+      ],
+      imports: [RouterTestingModule],
+    });
+
+    guard = TestBed.get(DeliveryModeSetGuard);
+    mockCheckoutService = TestBed.get(CheckoutService);
+  });
+
+  it('should redirect to deliveryMode page when no modes selected', done => {
```

Comment:
Add test for case when there isn't a step with type DeliveryMode. (Look at shippingAddressSetGuard)